### PR TITLE
Expose ultrasound timings

### PR DESCRIPTION
**Pull request by anton-firmware**

### CODE REVIEW

**Review by trickeydan — APPROVED** [review]

**Review by trickeydan — DISMISSED** [review]
Can we reduce teh code as discussed

File: src/firmware.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@ static const CommandResponse OK = "";
 // Multiplying by this converts round-trip duration in microseconds to distance to object in millimetres.
 static const float ULTRASOUND_COEFFICIENT = 1e-6 * 343.0 * 0.5 * 1e3;
 
-static const String FIRMWARE_VERSION = "SourceBots PWM/GPIO v2018.4.0";
+static const String FIRMWARE_VERSION = "SourceBots PWM/GPIO v2019.6.0";
```
Comment (Tyler-Ward) [inline src/firmware.cpp]: Version number has increased by two.
I assume this board wont have servo support any more so poosibly remove the PWM note.

File: src/firmware.cpp [hunk 2]
```diff
@@ -133,6 +133,43 @@ static CommandResponse servo(int requestID, String argument) {
   return OK;
 }
 
+static CommandResponse ultrasoundRawTiming(int requestID, String argument){
```
Comment (Tyler-Ward) [inline src/firmware.cpp]: other ultrasound function should call this function

File: README.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ Request IDs can be specified by prefacing your command with `@int`, for example
 | L       | Control the debug LED       | State {H, L} |                        |
 | R       | Read a digital pin          | Pin Number   |                        |
 | S       | Control a servo             | Servo Number | Width                  |
+| T       | Read a raw ultrasound timing | Trigger Pin  | Echo Pin               |
```
Comment (Tyler-Ward) [inline README.md]: description could be better e.g. Read a triggered pulse delay

**Review by Tyler-Ward — DISMISSED** [review]
looks good overall but could do with some cleanup and better code reuse

**Comment by Tyler-Ward** [comment]
Also CI is failing

File: src/firmware.cpp (hunk 1 above)
Comment (kierdavis) [reply src/firmware.cpp]: I'd assumed that the convention was `year.month.releasenumber`, but after looking at where the version number came from (f8bdaf6842d602b0cb9948ed347acb2067abd7a2) I have no clue.

Comment (trickeydan) [reply src/firmware.cpp]: > I'd assumed that the convention was `year.month.releasenumber`, but after looking at where the version number came from ([f8bdaf6](https://github.com/sourcebots/servo-firmware/commit/f8bdaf6842d602b0cb9948ed347acb2067abd7a2)) I have no clue.

That's what I thought.

@RealOrangeOne can you clarify for us?

File: README.md (hunk 3 above)
Comment (anton-firmware) [reply README.md]: fixed in 06541c9bfca36547e606e81bb8becaf70a3df4d0

File: src/firmware.cpp (hunk 2 above)
Comment (anton-firmware) [reply src/firmware.cpp]: it was decided to create an IO function that does the triggering and that would be called by both of the US functions.

File: src/firmware.cpp (hunk 1 above)
Comment (anton-firmware) [reply src/firmware.cpp]: updated in f8b5dc352b35eda7777af4309be8dcb2157f08ed

**Review by trickeydan — APPROVED** [review]

**Comment by trickeydan** [comment]
I've attempted to reduce code reuse further, but due to how the command handlers work, it actually isn't particularly easy. So this'll do I think

**Pull request merged**